### PR TITLE
fix(iOS): loader stuck on error of wallet binding

### LINKED ISSUE #2488: MOSIP VC Activation Fails for ED25519-Downloaded Credentials When Priority Is Set to 1 in Key Manager Settings (Android) and Gets Stuck on iOS
**Steps to Re-Produce :**
1. Open Inji wallet
2. Click on + Icon
3. Click on "Scan and Download Card"
4. Set the keys preference to "ED25519" from wallet application setting page
5. Download the Mosip VC 
6. Verify the VC Activation

 
Actual: 
1. ANDROID - VC activation is failing with error "We're sorry! Due to technical error, we are unable to server your request now. Please try again later!"
2. IOS - Getting stuck at "In progress" state. Use needs to close and reopen application to use it.

Expected: VC should get activated successfully. If keys are not supported user should get the valid error and he should be able to proceed with other flows/Actions.

Video:

https://github.com/user-attachments/assets/60578891-942c-4ed5-9fc9-76cdfcfc17b2
https://github.com/user-attachments/assets/a87e3d1f-6f46-44b5-85c1-ca971889f508

Android application logs:

[realme-RMX3371_mosipVCActivationFailure.log](https://github.com/user-attachments/files/29082360/realme-RMX3371_mosipVCActivationFailure.log)

Phone make/model:   Realme GT NEO 3T, Iphone 13
OS: Android 14, iOS 26.5
BLE version : 5.2, 5.0

Applications tested with:
Android - 242.apk
IOS - 0.23.0 (14)

### PR DESCRIPTION
**Pull request by KiruthikaJeyashankar**
## Summary by CodeRabbit

* **Bug Fixes**
  * Improved wallet binding overlays by delaying visibility on non-Android devices (0ms on Android) based on binding progress and error state.

* **New Features**
  * Enhanced overlay timing behavior through a more reliable “show after timeout” hook lifecycle.

* **Tests**
  * Expanded wallet binding test coverage for platform-specific timing and error edge cases.
  * Updated `useOverlayVisibleAfterTimeout` tests to match the revised timeout behavior.

* **Chores**
  * Removed native debug logging to reduce unnecessary stdout output.

Relates: https://github.com/inji/inji-wallet/issues/2488

### CODE REVIEW

File: screens/Home/MyVcs/WalletBinding.tsx [hunk 1]
```diff
@@ -25,6 +27,10 @@ export const WalletVerified: React.FC = () => {
 export const WalletBinding: React.FC<WalletBindingProps> = props => {
   const controller = useKebabPopUp(props);
   const {t} = useTranslation('WalletBinding');
+  const delay = isAndroid() ? 0 : 200;
+  const walletBindingInProgress = useOverlayVisibleAfterTimeout(controller.walletBindingInProgress, delay);
+  const isWalletBindingError = useOverlayVisibleAfterTimeout(!!controller.walletBindingError, delay);
+
```
[the remark this answers is not in the document]
Comment (KiruthikaJeyashankar) [reply screens/Home/MyVcs/WalletBinding.tsx:39]: useOverlayVisibleAfterTimeout has been designed for one state to other state transition delay, considering that should we still have the timeout ref defined?

File: shared/hooks/useOverlayVisibleAfterTimeout.test.ts [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
+    timeoutRef = {current: null};
 
-    let callCount = 0;
     jest.spyOn(require('react'), 'useState').mockImplementation((init: any) => {
-      callCount++;
-      if (callCount % 2 === 1) return [init, mockSetVisible];
-      return [init, mockSetSavingTimeout];
+      return [init, mockSetVisible];
     });
+    jest.spyOn(require('react'), 'useRef').mockReturnValue(timeoutRef);
```
[the remark this answers is not in the document]
Comment (KiruthikaJeyashankar) [reply shared/hooks/useOverlayVisibleAfterTimeout.test.ts:25]: This will cause error
```
Cannot redefine property: useState
TypeError: Cannot redefine property: useState
```

**Review by swatigoel — APPROVED** [review]

**Review by swatigoel — APPROVED** [review]
